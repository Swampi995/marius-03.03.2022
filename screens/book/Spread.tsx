import { FC } from 'react';
import { StyleSheet } from 'react-native';
import { Text, View } from '../../components';

interface SpreadProps {
    maxBid: number;
    minAsk: number;
}

const Spread: FC<SpreadProps> = ({ maxBid, minAsk }) => {
    const spread = (!minAsk || !maxBid) ? 0 : Math.round((minAsk - maxBid) * 10) / 10;
    const percentage = Math.round((spread * 100 / (maxBid || 1)) * 100) / 100;
    return (
        <View style={styles.container}>
            <Text style={styles.textLeft}>Spread: {spread.toFixed(1)}</Text>
            <Text style={styles.text}>({percentage.toFixed(2)}%)</Text>
        </View>
    );
}

export default Spread;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },
    textLeft: {
        marginRight: 6,
        fontSize: 14,
    },
    text: {
        fontSize: 14,
    },
});
