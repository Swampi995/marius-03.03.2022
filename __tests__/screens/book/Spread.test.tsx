import renderer from 'react-test-renderer';

import Spread from '../../../screens/book/Spread';

describe('<Spread />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Spread maxBid={5000} minAsk={5005} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
