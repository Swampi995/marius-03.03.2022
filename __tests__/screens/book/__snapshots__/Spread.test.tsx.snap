// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spread /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#121723",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#bebfc5",
          "fontFamily": undefined,
        },
        Object {
          "fontSize": 14,
          "marginRight": 6,
        },
      ]
    }
  >
    Spread: 
    5.0
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#bebfc5",
          "fontFamily": undefined,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    (
    0.10
    %)
  </Text>
</View>
`;
